release/models/aft/openconfig-aft-common.yang

```diff
@@ -385,6 +407,37 @@ submodule openconfig-aft-common {
     }
   }
 
+  grouping aft-common-entry-nexthop-gre-state {
+    description
+      "GRE encapsulation applied on a next-hop";
+
+    leaf src-ip {
+      type union {
+        type oc-inet:ip-address;
+        type string;
```

Comment:
If this is going to reference an L3 interface (I don't think it should, see other comment) then this needs to be an `interface-ref` which cannot be in this union, so will need a restructure to be able to reference an interface, subinterface tuple.

Reply:
> are leafrefs supported in unions now? I didnt find any existing examples of it.

All OC models are 1.0 based - this is dependent on a entire model-set shift to 1.1+

Reply:
> > are leafrefs supported in unions now? I didnt find any existing examples of it.
> 
> All OC models are 1.0 based - this is dependent on a entire model-set shift to 1.1+

Thanks for confirming this. As per @robshakir feedback (`Given that this is state, can we not just simplify this down to only using the IP address? The implementation can resolve what address is being used if needs be.`) I have made this IP address only